### PR TITLE
Make Ospec Flems-friendly

### PR DESCRIPTION
**Pull request by pygy**
There are a few annoyances when using Ospec in Flems:

1) It only works in CommonJS environments
2) It uses console.error, which in Flems prints out the full stack trace (which can be long in Ospec thanks to nextTickish)

3) It expects to find its own file name in the path (to clean up stack traces), but Flems only provides a hash, that's more of a Flems issue probably.

This fixes 1 and 2

edit: and now 3, and also #2036

### CODE REVIEW

**Comment by pygy** [comment]
Ok, Gandalf :-)

**Comment by pygy** [comment]
Please note that the second commit is possibly a breaking change since it causes the report to always run async where it was possibly synchronous in the absence of an async test. Not sure if it matters...

**Comment by tivac** [comment]
Yeah, that's a breaking change. `ospec@2` here we come XD

**Comment by pygy** [comment]
@tivac: contrast [this](https://flems.io/#0=N4IgtglgJlA2CmIBcBWALAOgGwEYA0IAZhAgM7IDaoAdgIZiJIgYAWALmLCAQMYD21NvEHIQIAL54a9RswBW5XgKEim-aqTYACAE7Co8HVoC8WsBj3UDOgDrV1m3Xz7bTUPjwCuDQRgBGfFAAnnZ2fP7whHx6AKK0PCwAFIkAlMYAfMB2Wk4uGEIAHmwAwsrCrloA5JXZWgYIQrlsGABu1IHwpHbiKaHtiTYgAnpgfC3wdXyd1JXaxHpaAlrtBqRabCy02gDmfFqEOnxgWgAOzrDre2d8F4kAxDgAnI8ADL3c+572bBACqVpZag5Fq0IzDeCjcYmRYYUgnIKpADcfRylmsiUOLjwWgoYESlSgEBalWxeIJRMqKRS2LM+MJxNJdIp2OA4Mh8CQi2oIzG8B6KQAuilkUDdPpDBjnGxsbimQzaeTiVShSLUeKdJKsTiVSjFok2byMDxaLBYKUvmwUhh4ABHTwm0iJHC9ag9MIDIbciG8ybTWb7CALADuLGEYshEGo23Woa0CRIUEsi0IWloyw6Ma2lTWfkOnm27C0fniAGt9odjhsJtdYINsYQvjwfn8UgDaiCwV72dDwnCES7gaCtG0M6ZgGxaNtOYN6XWtCX4EFOfg4ywE5ZOVQJ1OtDOiXOtmwdKROayu7zOQbxuJxALxO2h0IwCdoePJ9OQLOPgul1oAEz3qKHZaJ4JxQFs8BQK+24fl+2I-su2LxrAibCJub47oMJyDLegG1GiEqYtKOIjgYQr4eqmrERQT4nORooERqREyqB4FCFA9G1HwiSkfAGDuGAVq2vasCOqxEFQPxRwDnqV58caprmoIQl2g6Toum6-QzgI-q0DARallo8AID4bBrAcRzlpOplrBsWypgsuZ8PmhbFjwZYWZWsY1loDzPDgc4Nt8vzUP8gJqlYhFSjKtQ5BhH4UHOyGodQm53ngsUAjBu4gIlHzJRu2qATk9EReizE4pl8U5XlSFrihhUUOlu6inF2WDLVq7rmhOLVXuLQ4XetSlWKkVMdFlWtVl741Ul9UpZu1CeKaQ2iiNjFUTFU19blc3dalRUZdt7W7fl82NatJUyVxVFGvNAByHSkBgCBRhsKkiY6bzdC63HaTM2h6VBbllsZELlOZFZWdsNmZoDjnwJG0Y8uMUF+Y8AUfEFTYhWFFFjZtk05G1M0dXtDU9U1kgtcT02YSA2ESEdV34+VE2JcdpOnXV+3oSdcHrPARSciguHDTJG0VRztM7Z1BWU+lVUnXL50K8VWgjTdRF3Qmj2rC9wjbO91qqaJiTfa6v0epsVhkMsy0XCCsCeJ0WiRiB1A-pBWiwBAmhrHwKaEoQhCGOUPuGxsgWNs2oWtuFw5DgIPB6BBPawvCSIPp2KMTKYvaZ8K2dcuJjT5xn-aqp8wVLIex541NehsJ4OhAmeKfwBBl72KnQjYnJ3e5-3S1gV3Jej0I6t4QxlFSztAt1yeqZsEejopNTS0rUXM8E1Lm+wCy-P7h8i+covqQbw7OrUDdye9-JJpms5ykm59iR-jJ3FyUaj9KZar9qWdLfEebEH6KWfv-YSakLaaT+p+HScx4BsASDGCY1AhZsAAMoQD8L7KM5ZLJVlOOcaONc45th3mzLUW4uYq15r1I+A1vyLmXNTeex94IsP-LhQ+M1KgnEqLeCWs92ay3JgtQ6dNOT8MEetER1CxFnXoTQ+mAsELcN4aojh84uE4B4VIqoAihG1Buk1DAYBaAnB-qaW6yU9adHrDHXGxl45NxbkCYyGAVjwHujIfk-F4DwBODEU2joKCVAACIAEkABqJIqjRLidiSoAAFSo9FYEei9lBY0CRvZg1hqCCYegAC0kYIA-BNBAAAXt7EMYYnIuW0CDAhXlqznC0PcP8LwXgAGZ3iOLIZ0+OxcBClzzjCPsqRRk907mXSZhdi65NDFBMcmUToACE5zqJXPLA60taYGMGAAQQPCvY8p475zI5FyDuEFh7jO7uM6mhQ2AflORITKApajq0lqIxh4jGrLMgneORu92ZgqodRdhTCeYU32cCjixib6ii-rMiC1in4Wg+mpD+wDxmYr-jis2MCXTukGF8Rc3tEVGRMhDByxT4BlOoBUiAVTalQXqUCRpBZmmGU8qg4hNxOl3G6X0gZ1ccZLAboOTs4z05TJksBK5ady6KqWfEFZr51lcy2Uo+Fm5Mok3ph87Ep8AQqqEN3O5fdx6gKeRPPk2JXnvJwl8n5rMooKIBfqiRFBEWguEeC6hkKvXQp9XCv1AbkXAJtWArFL8oFmzxai-UICMUKQTZA0J5sNJkq0iAbJcZNX5LpYINYRSxTMtZeyupoZuV5l5QZdyrTlidHYkK0SIqxX9NIVKoEMrE5ysdQqxZQEk7ovmQXfsGq8mrIobTTZ2zdGRsakao5IBTXL1XpcydNzLXwAeY6h1oCXkYNdZ8qa3zRS-PkeGrmAs9mbmjaG8a3qH3aKfUVV9hMmpBqhTKGFgLKYvvzTkNFcbCUQOJY6FN4G00EszUSgBJK819DgV8ItNKCn0sraU8plTfYcq0FyosjbXL8qhugzQ3saxrC6T03tWMnHSpGeO4doDR0zvY7c++XHpk8ZpWsqaS7fVrqmsai9ZrzlL3bvfa198j32rtRBM9wscofPViVD1lCw2AYjV1A1OJQOerfferRsLDN+sDaZ39P654Ga-f6ktSLOKpoPVB7FKHYOfwQ46zziac2kvQxYL4qQmbSAYKII0pBFAgHUCoNgogJACgIHgks5AkBUBAHQKLTA+BwngDwbgIAW5cCYOwNgJwTwAHoatfBOCWbYRojg1YKycIrAABHAGBMAvDa4VngGAFAlbYEEDrogRsEBOFsPJmWKALZeAQNggw+ggFSxQEp+BwCBGWnxIWZwdBmXW3gCg23lUnYoEtkA0JqAnb-H+PAOBel4BQI8V7jwBTpUi7ISAGwdAkG6xgHrKAStldEJV6rSA6sNaay1sANW-ssAB7AIHIPEcVORyQYbcWxsTaYCN3CQA) (vanilla ospec) with [this](https://flems.io/#0=N4IgtglgJlA2CmIBcBWALAOgGwEYA0IAZhAgM7IDaoAdgIZiJIgYAWALmLCAQMYD21NvEHIQIAL54a9RswBW5XgKEim-aqTYACAE7Co8HVoC8WsBj3UDOgDrV1m3Xz7bTUPjwCuDQRgBGfFAAnnZ2fP7whHx6AKK0PCwAFIkAlMYAfMB2Wk4uGEIAHmwAwsrCrloA5JXZWgYIQrlsGABu1IHwpHbiKaHtiTYgAnpgfC3wdXyd1JXaxHpaAlrtBqRabCy02gDmfFqEOnxgWgAOzrDre2d8F4kAxDgAnI8ADL3c+572bBACqVpZag5Fq0IzDeCjcYmRYYUgnIKpADcfRylmsiUOLjwWgoYESlSgEBalWxeIJRMqKRS2LM+MJxNJdIp2OA4Mh8CQi2oIzG8B6KQAuilkUDdPpDBjnGxsbimQzaeTiVShSLUeKdJKsTiVSjFok2byMDxaLBYKUvmwUhh4ABHTwm0iJHC9ag9MIDIbciG8ybTWb7CALADuLGEYshEGo23Woa0CRIUEsi0IWloyw6Ma2lTWfkOnm27C0fniAGt9odjhsJtdYINsYQvjwfn8UgDaiCwV72dDwnCES7gaCtG0M6ZgGxaNtOYN6XWtCX4EFOfg4ywE5ZOVQJ1OtDOiXOtmwdKROayu7zOQbxuJxALxO2h0IwCdoePJ9OQLOPgul1oAEz3qKHZaJ4JxQFs8BQK+24fl+2I-su2LxrAibCJub47oMJyDLegG1GiEqYtKOIjgYQr4eqmrERQT4nORooERqREyqB4FCFA9G1HwiSkfAGDuGAVq2vasCOqxEFQPxRwDnqV58caprmoIQl2g6Toum6-QzgI-q0DARallo8AID4bBrAcRzlpOplrBsWypgsuZ8PmhbFjwZYWZWsY1loDzPDgc4Nt8vzUP8gJqlYhFSjKtQ5BhH4UHOyGodQm53ngsUAjBu4gIlHzJRu2qATk9EReizE4pl8U5XlSFrihhUUOlu6inF2WDLVq7rmhOLVXuLQ4XetSlWKkVMdFlWtVl741Ul9UpZu1CeKaQ2iiNjFUTFU19blc3dalRUZdt7W7fl82NatJUyVxVFGvNAByHSkBgCBRhsKkiY6bzdC63HaTM2h6VBbllsZELlOZFZWdsNmZoDjnwJG0Y8uMUF+Y8AUfEFTYhWFFFjZtk05G1M0dXtDU9U1kgtcT02YSA2ESEdV34+VE2JcdpOnXV+3oSdcHrPARSciguHDTJG0VRztM7Z1BWU+lVUnXL50K8VWgjTdRF3Qmj2rC9wjbO91qqaJiTfa6v0epsVhkMsy0XCCsCeJ0WiRiB1A-pBWiwBAmhrHwKaEoQhCGOUPuGxsgWNs2oWtuFw5DgIPB6BBPawvCSIPp2KMTKYvaZ8K2dcuJjT5xn-aqp8wVLIex541NehsJ4OhAmeKfwBBl72KnQjYnJ3e5-3S1gV3Jej0I6t4QxlFSztAt1yeqZsEejopNTS0rUXM8E1Lm+wCy-P7h8i+covqQbw7OrUDdye9-JJpms5ykm59iR-jJ3FyUaj9KZar9qWdLfEebEH6KWfv-YSakLaaT+p+HScx4BsASDGCY1AhZsAAMoQD8L7KM5ZLJVlOOcaONc45th3mzLUW4uYq15r1I+A1vyLmXNTeex94IsP-LhQ+M1KgnEqLeCWs92ay3JgtQ6dNOT8MEetER1CxFnXoTQ+mAsELcN4aojh84uE4B4VIqoAihG1Buk1DAYBaAnB-qaW6yU9adHrDHXGxl45NxbkCYyGAVjwHujIfk-F4DwBODEU2joKCVAACIAEkABqJIqjRLidiSoAAFSo9FYEei9lBY0CRvZg1hqCCYegAC0kYIA-BNBAAAXt7EMYYnIuW0CDAhXlqznC0PcP8LwXgAGZ3iOLIZ0+OxcBClzzjCPsqRRk907mXSZhdi65NDFBMcmUToACE5zqJXPLA60taYGMGAAQQPCvY8p475zI5FyDuEFh7jO7uM6mhQ2AflORITKApajq0lqIxh4jGrLMgneORu92ZgqodRdhTCeYU32cCjixib6ii-rMiC1in4Wg+mpD+wDxmYr-jis2MCXTukGF8Rc3tEVGRMhDByxT4BlOoBUiAVTalQXqUCRpBZmmGU8qg4hNxOl3G6X0gZ1ccZLAboOTs4z05TJksBK5ady6KqWfEFZr51lcy2Uo+Fm5Mok3ph87Ep8AQqqEN3O5fdx6gKeRPPk2JXnvJwl8n5rMooKIBfqiRFBEWguEeC6hkKvXQp9XCv1AbkXAJtWArFL8oFmzxai-UICMUKQTZA0J5sNJkq0iAbJcZNX5LpYINYRSxTMtZeyupoZuV5l5QZdyrTlidHYkK0SIqxX9NIVKoEMrE5ysdQqxZQEk7ovmQXfsGq8mrIobTTZ2zdGRsakao5IBTXL1XpcydNzLXwAeY6h1oCXkYNdZ8qa3zRS-PkeGrmAs9mbmjaG8a3qH3aKfUVV9hMmpBqhTKGFgLKYvvzTkNFcbCUQOJY6FN4G00EszUSgBJK819DgV8ItNKCn0sraU8plTfYcq0FyosjbXL8qhugzQ3saxrC6T03tWMnHSpGeO4doDR0zvY7c++XHpk8ZpWsqaS7fVrqmsai9ZrzlL3bvfa198j32rtRBM9wscofPViVD1lCw2AYjV1A1OJQOerfferRsLDN+sDaZ39P654Ga-f6ktSLOKpoPVB7FKHYOfwQ46zziac2kvQxYL4qQmbSAYKII0pBFAgHUCoNgogJACgIHgks5AkBUBAHQKLTA+BwngDwbgIAW5cCYOwNgJwTwAHoatfBOCWbYRojg1YKycIrAABHAGBMAvDa4VngGAFAlbYEEDrogRsEBOFsPJmWKALZeAQNggw+ggFSxQEp+BwCBGWnxIWZwdBmXW3gCg23lUnYoEtkA0JqAnb-H+PAOBel4BQI8V7jwNtXYIE1bEl3tv-AyFoUgSCAAqEAGDOTYIkS7129CHctP9ggKRYcED6DYFbSXUs4BQHgXpL2cc4CW4Tl4Ap0qRdkJADYOgSDdYwD1lAJWyuiEq9VpAdWGtNZa2AGrVOWA09gHThnvOKn85IMNuLY2JtMBG7hIAA) (with setTimeout)...

For the third issue, there may be a way on our end to detect the path of the current file but it won't be trivial: 

Chrome:

`'    at new init (blob:https://r.flems.io/fec76e22-06fc-49ab-9da0-463dd47d7a6d:5:14)'`

Firefox:

`'init@blob:https://r.flems.io/88cd211c-dfe2-1344-8b6a-1c2bd3023f61:5:14'`

Safari: 

`'init@blob:https://r.flems.io/72f1d451-db1d-4535-8923-6c1a70229193:5:23'` (like Firefox, but with a different column)

Node:

`'    at init (/Users/pygy/src/mithril.js/ospec/ospec.js:7:13)'` (after the UMD-ish addition, no `new init` anymore as IIFE)

this was putting the throw() inside the `init` factory, but when called as `o.new`, it gives yet another trace entry (Node):

`'    at Function.init [as new] (/Users/pygy/src/mithril.js/ospec/ospec.js:7:13)'`

If IE/edge have similar traces, we may have it as the first line that matches `/\/(.*?):\d+:\d+/`...

**Comment by pygy** [comment]
It seems to [work so far](https://flems.io/#0=N4IgtglgJlA2CmIBcBWALAOgGwEYA0IAzvAgMYAu8UyIAFueQA6FID0rArgHaMDWA5hlIB7MK2GFG8UgAEcGTAAZxk6RgBWhEAQBmEBFqQBtUFwCGYREhAZ6YWNpAiulFzRABfPKYtWbmx2dXchpnQnIAAgAneC4oeCiIgF4IsAwYuISAHS4wyKjhYUiUqGFSDksXDAAjYSgATxyc4Rr4HWEYgFEzUloACj6ASiSAPmAciOjC8gxKAA9yAGFhF1jiiIByDYmI+IRKKaKMADcuOvhCHI9BprO+rJAVmLBhY-hd4QuuDci9GIiVhEzvFCBFyLQzJF+MIIjoCmAIoxCrAwTCkcIUX0AMQ4ACcuMUN20sO4FAgKyGEXGXEmxzMiSe8Beb2SAIwknqQwA3LdJhl4lE+gUingIkYwH0NlAIMcNqKJVKZRtBoNRalJdLZfKNUrRcBGcz4EgAVxnq94NdBgBdQY8mnRWICoXTUXinVa9WK2Uqm12vmOhLOkVi328gF9A3moRmWCwZbcciDDDwACOHBjhD6OBuXGuzXuj1NTPNHy+P1hEH+AHdaLEHcyIFx+GDaxFevooBkAToImYgecW5CNqDqgUOPx6BFqj1eLD4S33ujYA9RTpSeRyVxKdTafSTWaWSkWhyhjs6YlTgOUsByGZ+MaHpqVxFePB6sb8G3aB2MsaTLf7wiR8ZWfSFyCiFgqUjN5jWgi0PCtDwzz3SgwEYVkbzvB8QCfYlX3fCIACYkPtc8Ig4RgoEhKgMIA7DcNFfCP1FdtYE7WI-0wwCHkYB4EJInZ+UDYVyFdS94htQSA0FETXVQxhJPtISZJdMUKKoygoEUnZhD6cT4AwUowCTVN01gTN1OoqBDNEHNJl0uDo1jeMXBMtMMyzHM8zuR8VnLMwYCnGcIhIJk1lBOFRDnO9KnIUFwUhXt-lHYRx0nadSFnSKEXBRdkQiHF8RwZ811yDcKUGKkpMyFTgyMHZJi47D6uJVj2K4P9ELwBqqTooCQBaljvzY38QxIyZFP9Gqg1EsUeqa-rBq-H8OLG0V5r6h4lra0b-yw-rcP4nZJodabZLm+1Gs2gbnx21ajC4DhY0Q467NOp1zvqy7ev2rbbuG9rOq8Dbfpu1qAd2l77ROnSZqEAGADlzkIDAECbcE3LMzNCSuHNdN875IgCqAgsykKEFiiL5zhGLwsHInkvgRtmwPGjCtxYriVKskKqqpTpJm10Qe4sGhpWjq1qA76Fp4vjuuht7lMFi7Jiu0Htoh+6ZZwkDiXmchjRQI6Feqj7VK+1WfpFjXxaB+XLe1m2RvuqGJre2GRPhjskZBVHYn4DHk3c8y+hx3M8YLCE4gMIEnpROlYA4C4IkbciuHwmjYAgcJQWEHtpR0HQEjWCI0YD2gSvXTdt2QhlchiajWWPRhOVtWv92LQ82RPNvSL3FZLIOI92Rb7kdm58qaTAiCa++mJyA4KIaX1ev4Go2DV+o0U4I31nt8eyj15NQeLR2AT+bO83tYY3sGAg41p8zQYvFj57e6ms26se2M9Wum-H4fnfJ+L9v6wFDFwWGKxSAN0oE5OMqVXJByxn0Qib0HJFkNHAlyiYkEeWzJAg+GkDKkBjPAhMmMPJh28vjHCflfjwHIL0BcQJ4ALAAMoQGqFnJsc4oq5URMiSuZVq6VR3O9YSV9rpO0BmKa+utGJvg-C-ORxxnxMSIvxX++0NiMA2AhRWAtPqO3+rbSWC0dF6JOkrIxUiTHOwlntEWN91HES0U4+RL5FERBwJoq2xoLH6J2LDIwVoMBgDMIwOBcM2o+wuKuKuFISCiPnovGkJAMDAngAjXwlpDLwHgIwTowdMxGA2AAEQAJIADU5SbEqTU0UGwAAKGxFLUILBnEmJDeg0VCpTJK7wYgAFpGwQA3DGCAAAvGiNY6wpTSpEDKWV5z8KXBEbEhFFCKAAMxEnicIwEs9dx1xPk3Eerd25QJge8YePd27dNrCTa8PVroACE1FeM-HdBxPU1YiwAIKgSAcaFe0C16UA3mCrex9D4QphUQl++tsKAs8D1K0Z9TYSLqiouxMijAPKoIhKxhjzbEsvtiv+Hjvl-gJVpQJED7ToKhbAkhzkEE4NMh5VBBCT5YPZRQkOVCcz5geNwN8NFaXkzCi4UE9JBnwBGVwMZEAJnTJJrMmk8yJyLOCtlZhayNlbN2UInmNIjkRDIgPWFNzu6jzepazeQ9bUXL7okSVzzvpvNxbtX5fj+ootFAAqCjqjQmmZfAfeJ8N4n0Rawg2-q+JooxRfT+s1HH0SpZrBxtKiUGPJWmslqbXQ4vBqY-FPRHm5rDEy65fLyG4JDtyxlEZCHUTrYgzlgqvLCp8iATpbYK29IpnTOVDpFXKtVTM2smqxzatJssqKXALiaQERiUEhqdl7JJAcs1ojLmtqdc3F1xyw3XLOXc11A6elPL5pbL1eFPli3sX+X12sA233ApBUF1zIXXMjda6N1rY0LGRYm766L7TjXEbVNNJan14pzfSj+WLYOUtUaW59Y1C0oddNhmDxa0Pevuoh7SzarnguIaQ7BArMxNvsi23lrKyEduKaHbttwaHcH7ZKvpI7-jDNGeMrOaqIgaqnLO9Kur5xLvCDRJc66sSbM3Saye6y92XqtUQ89dr93hu08ei1e53W3tVve+DPrvp-NA8SIN36KO-oo-+ohgGEWiiRQm1F4Hk3Ifw7IwjGGEODrpXh5W6aDqZrLVWlNOGQx5qLX50GN9qVihIz2+j5G21Meow22jaCGPWvbRy1jQqOPpG4EMTw3gQDmEsDQIQhAtAECCGsdwCECDcN4IYEw1XfA0AkFIUgjhF4OGsPQJgLB2DcD4IIEQYh+vSDkAoDAyh5ukA0I1kA5B6hSBoAEAgjBIQ9K60YIwigCDkAeLcEAVo8BGCGfgcAdQnoGVYUiKIcVru3Ye5az7p2CCsi4J9wihE8A4G2XgFAuIIe4huyds7PXLDqd+w9rId96i9VoAUKsLDsedCiAUKIHgB2MNoOs+A6nUeo7IuEGcAAVKIPR4AAFlDu1kSCkJd2OABK8B+CdDmIwe4ECLsi4eAAPT6AAfiQFLgAhHjgnAAfS7wvKfkAANQhXZLeTKYTWd9FYFkLIrA+gYAAFSS8GEgI3UB1fW6yLb1ggwjA4Ag2r1HmuHhJkGGbgAJM+B4-AwAq7V8K8gJFfvw6p3uVQn2cBYFBwAdhQHgAAHFD9PsO-sDRuxEZHBBKSjAiMQcgtOICWFSuQPokeCAxDe4mfPIBBg15ALcNXceU-bPB-HrAUOe8w9u9nkn1fYf3YIKwCX0u5cK46Ir+qxujdG6tKt+fhvF-L9UKQI363vd+9YGARvNPMr08ZyzknCQW9h9R59lAKfCIp8T2gNA+BH-P6tF1HwtXrCQHBFEfQi35AUAhsogRs6AGBmA2BOAeABAhBRB98xlMd-95BAD4Df99B1tHAtsdtrAAh+IgA) (Node/Firefox/Chrome/Safari).

Could someone try the flems above in IE/Edge and report if the error log points to the right lines?

File: ospec/tests/test-ospec.js [hunk 1]
```diff
@@ -183,18 +187,6 @@ o.spec("ospec", function() {
 		})
 	})
 
-	o.spec("stack trace cleaner", function() {
```
Comment (pygy) [inline ospec/tests/test-ospec.js]: The new cleaner expects that the stack comes from within ospec, with at least one framework function on the stack before we reach user code, breaking this test. I've thus moved it to the new reporter test that provides real ospec stack traces.

**Comment by pygy** [comment]
Actually, for the stack trace cleaner, starting from the bottom of the stack would be better still but that's for tomorrow.

**Comment by pygy** [comment]
Aargh, that doesn't work either in some scenarios...

**Comment by pygy** [comment]
At last, I think this is reliable.

I changed the `error` field in the report function to point to the error object rather than just the stack trace. the undocumented `o.clearStackTrace()` now takes an error object rather than a stack trace string. So code that was using `errorLocation = o.clearStackTrace(e.error)` will still work, but other uses of either `clearStackTrace` or `e.error` will break obviously.

Pinging @zyrolasting who introduced that change.

**Comment by zyrolasting** [comment]
I understand the lack of love for the breaking change only because adding the reporting function meant making all results part of the public API. I do support adding the full error object if only to improve `source-map-support` integration.

**Comment by pygy** [comment]
@zyrolasting changing `e.error` isn't technically a breaking change because your patch hasn't been released yet on NPM. I gave you the heads up about the update because I supposed that you were already relying on the API introduced there in your own code.

The breaking change is the `o.run()` change from [zalgo](http://blog.izs.me/post/59142742143/designing-apis-for-asynchrony) to always async.

Before, if none of your tests were async you could do

```JS
o.run()
console.log("printed after the report")
```

Not anymore after this PR.

We may want to provide `o.report`

to enable

```JS
o.run(results => {
  o.report(results)
  console.log("printed after the report")
})
```

**Comment by pygy** [comment]
There, now the reporter is exposed as o.report(results) and it returns the number of errors.

I almost moved the `process.exit(1)` call to the binary, but then I realized that doing so would prevent extra instances created by `o.new()` from exiting on error by default. Enough breakage already...

**Comment by pygy** [comment]
I think I'm done :-)

**Pull request merged**